### PR TITLE
Add detached payload for mac

### PR DESCRIPTION
**Pull request by kentakayama**
Add "detached content" mode for COSE_Mac and COSE_Mac0.
The interfaces of `t_cose_mac_encode_parameters` and `t_cose_mac_encode_tag` are changed like `t_cose_sign_encode_start` and `t_cose_sign_encode_finish` respectively.

### CODE REVIEW

File: test/t_cose_compute_validate_mac_test.c [hunk 1]
```diff
@@ -347,4 +347,228 @@ int_fast32_t compute_validate_get_size_mac_test()
...
+    QCBOREncodeContext           cbor_encode;
+    int32_t                      return_value;
+    enum t_cose_err_t            result;
+    Q_USEFUL_BUF_MAKE_STACK_UB(  signed_cose_buffer, 300);
+    struct q_useful_buf_c        signed_cose;
+    struct t_cose_key            key;
+    struct q_useful_buf_c        payload;
+    QCBORError                   cbor_error;
+    struct t_cose_mac_validate_ctx verify_ctx;
+    size_t                       tamper_offset;
```
Comment (laurencelundblade) [inline test/t_cose_compute_validate_mac_test.c:366]: tamper_offset is unused and should be removed.

**Review by laurencelundblade — CHANGES_REQUESTED** [review]
Looks really good! Will merge with the one-line fix for the unused variable.

I'm pretty much convince that #122 is a good idea for the simplicity and code size reduction even though it makes calling a little more complex.

Thank you!

LL

File: test/t_cose_compute_validate_mac_test.c (hunk 1 above)
Comment (kentakayama) [reply test/t_cose_compute_validate_mac_test.c:365]: New commit [0b3db33](https://github.com/laurencelundblade/t_cose/pull/143/commits/0b3db3346b020a17adbb9098ca0dfb0d9678bae2) fixed this, and now `$ make -f Makefile.psa CMD_LINE=-Wall` shows no warning.

File: test/t_cose_compute_validate_mac_test.c [hunk 2]
```diff
@@ -135,7 +135,6 @@ int_fast32_t compute_validate_mac_sig_fail_test()
     struct q_useful_buf_c        payload;
     QCBORError                   cbor_error;
     struct t_cose_mac_validate_ctx verify_ctx;
-    size_t                       tamper_offset;
```
Comment (laurencelundblade) [inline test/t_cose_compute_validate_mac_test.c:138]: I think I marked the wrong tamper_offset as unused. Didn't realize there were two. 

This one is needed and the one below is unused. 

It compiles OK for OpenSSl, but not for PSA.

Sorry for the confusion.

**Review by laurencelundblade — CHANGES_REQUESTED** [review]

**Pull request merged**